### DISCUSSION TITLE
0.1.0-rc3 has been released!

### DISCUSSION THREAD

**Discussion by mihe** [godot-jolt/godot-jolt#387] [Announcements]
This is the third **release candidate** for version **0.1.0**, and assuming no critical bugs is **likely to become 0.1.0-stable**.

This release candidate supports **Godot 4.0.3**.

## Known major bugs

- Extension classes cause exception when used from C# ([#270](https://github.com/godot-jolt/godot-jolt/issues/270))
- Changing the `recovery_as_collision` parameter makes no difference ([#337](https://github.com/godot-jolt/godot-jolt/issues/337))

## Changelog

With there being no stable version to break yet, this release candidate takes the opportunity to include a few such changes.

### Removed

- ⚠️ The project setting **More Deterministic** under **Simulation** has been removed. It's now hardcoded to `true`.

### Changed

- ⚠️ The project setting **Recovery Speed** under **Kinematics** has been renamed to **Recovery Amount** and is now a percentage instead of a factor
- ⚠️ The project setting **Stabilization Factor** under **Solver** has been renamed to **Position Correction** and is now a percentage instead of a factor
- The project setting **Contact Speculative Distance** now has a smaller step size ([#372](https://github.com/godot-jolt/godot-jolt/issues/372))
- The project setting **Contact Allowed Penetration** now has a smaller step size ([#372](https://github.com/godot-jolt/godot-jolt/issues/372))

### Added

- There's now a project setting called **Movement Threshold** under **Continuous CD** for fine-tuning CCD behavior ([#374](https://github.com/godot-jolt/godot-jolt/issues/374))
- There's now a project setting called **Max Penetration** under **Continuous CD** for fine-tuning CCD behavior ([#374](https://github.com/godot-jolt/godot-jolt/issues/374))

### Fixed

- Opening a .NET project with Godot Jolt present should no longer crash on Linux ([#373](https://github.com/godot-jolt/godot-jolt/issues/373))
- Linux support should now be more comprehensive, covering distributions like Ubuntu 20.04 (Focal Fossa)
- macOS support should now be more comprehensive, covering versions as far back as 10.12 (Sierra)

<hr /><em>This discussion was created from the release <a href='https://github.com/godot-jolt/godot-jolt/releases/tag/v0.1.0-rc3'>0.1.0-rc3</a>.</em>

**Comment by jamesdimick** [comment]
Thanks for the fix for Linux, now I can at least open my project. :)

Although I'm getting a lot of error spam when trying to run my test scene. And the errors don't really seem to point to anything that I can fix...

```
E 0:00:00:0907   ScriptManagerBridge.cs:263 @ System.Type Godot.Bridge.ScriptManagerBridge.<TypeGetProxyClass>g__GetTypeByGodotClassAttr|12_0(System.Reflection.Assembly , System.String ): System.NullReferenceException: Object reference not set to an instance of an object.
  <C# Error>     System.NullReferenceException
  <C# Source>    /root/godot/modules/mono/glue/GodotSharp/GodotSharp/Core/Bridge/ScriptManagerBridge.cs:263 @ System.Type Godot.Bridge.ScriptManagerBridge.<TypeGetProxyClass>g__GetTypeByGodotClassAttr|12_0(System.Reflection.Assembly , System.String )
  <Stack Trace>  ScriptManagerBridge.cs:263 @ System.Type Godot.Bridge.ScriptManagerBridge.<TypeGetProxyClass>g__GetTypeByGodotClassAttr|12_0(System.Reflection.Assembly , System.String )
                 ScriptManagerBridge.cs:257 @ System.Type Godot.Bridge.ScriptManagerBridge.TypeGetProxyClass(System.String )
                 ScriptManagerBridge.cs:101 @ IntPtr Godot.Bridge.ScriptManagerBridge.CreateManagedForGodotObjectBinding(Godot.NativeInterop.godot_string_name* , IntPtr )
```

```
E 0:00:00:0908   Godot.NativeInterop.NativeFuncs.generated.cs:129 @ IntPtr Godot.NativeInterop.NativeFuncs.godotsharp_internal_unmanaged_get_instance_binding_managed(IntPtr ): Parameter "strong_gchandle.value" is null.
  <C++ Source>   modules/mono/csharp_script.cpp:1224 @ setup_csharp_script_binding()
  <Stack Trace>  Godot.NativeInterop.NativeFuncs.generated.cs:129 @ IntPtr Godot.NativeInterop.NativeFuncs.godotsharp_internal_unmanaged_get_instance_binding_managed(IntPtr )
                 InteropUtils.cs:35 @ Godot.GodotObject Godot.NativeInterop.InteropUtils.UnmanagedGetManaged(IntPtr )
                 VariantUtils.cs:464 @ Godot.GodotObject Godot.NativeInterop.VariantUtils.ConvertToGodotObject(Godot.NativeInterop.godot_variant& )
                 VariantUtils.generic.cs:385 @ T Godot.NativeInterop.VariantUtils.ConvertTo<T >(Godot.NativeInterop.godot_variant& )
                 RigidBody3D.cs:1117 @ Boolean Godot.RigidBody3D.InvokeGodotClassMethod(Godot.NativeInterop.godot_string_name& , Godot.NativeInterop.NativeVariantPtrArgs , Godot.NativeInterop.godot_variant& )
                 CSharpInstanceBridge.cs:24 @ Godot.NativeInterop.godot_bool Godot.Bridge.CSharpInstanceBridge.Call(IntPtr , Godot.NativeInterop.godot_string_name* , Godot.NativeInterop.godot_variant** , Int32 , Godot.NativeInterop.godot_variant_call_error* , Godot.NativeInterop.godot_variant* )
```

I've pretty much commented out most of my code at this point trying to find any issues on my end. Basically the only code left that even does anything with the physics system is a simple call to `bodyState.ApplyCentralImpulse()`.

**Reply by mihe** [reply]
> now I can at least open my project. :)

One step at a time, I guess. :)

The errors you're seeing are likely related to the issue listed in the release notes: [#270](https://github.com/godot-jolt/godot-jolt/issues/270)

C# support for GDExtension stuff is pretty broken at the moment, and any override of `_IntegrateForces` as well as any usage of `PhysicsServer3D`, `PhysicsDirectBodyState3D` or `PhysicsDirectSpaceState3D` will break things unfortunately. I realize this can make the whole thing a non-starter for some, but my hands are tied until [my pull request](https://github.com/godotengine/godot/pull/75955) gets reviewed and (hopefully) merged.

**Reply by jamesdimick** [reply]
> > now I can at least open my project. :)
> 
> One step at a time, I guess. :)
> 
> The errors you're seeing are likely related to the issue listed in the release notes: [#270](https://github.com/godot-jolt/godot-jolt/issues/270)
> 
> C# support for GDExtension stuff is pretty broken at the moment, and any override of `_IntegrateForces` as well as any usage of `PhysicsServer3D`, `PhysicsDirectBodyState3D` or `PhysicsDirectSpaceState3D` will break things unfortunately. I realize this can make the whole thing a non-starter for some, but my hands are tied until [my pull request](https://github.com/godotengine/godot/pull/75955) gets reviewed and (hopefully) merged.

Ah, I missed that. Yeah, hopefully it happens soon. :)
